src/picotorrent/persistenttorrentlistview.cpp

```diff
@@ -1,128 +1,130 @@
-    }
-
-    if (RestoreValue("Col_CompletedOn_Width", &colWidth))
-    {
-        m_tlv->GetColumn(TorrentListViewModel::Columns::CompletedOn)->SetWidth(colWidth);
-    }
-
-    if (RestoreValue("SortIndex", &sortIndex)
-        && RestoreValue("SortAscending", &sortAscending))
-    {
-        m_tlv->GetColumn(sortIndex)->SetSortOrder(sortAscending);
-        m_tlv->Sort();
-    }
-
-    return true;
-}
-
-void PersistentTorrentListView::Save() const
-{
-    if (m_tlv->GetSortingColumn() != nullptr)
-    {
-        int sortIndex = m_tlv->GetColumnIndex(m_tlv->GetSortingColumn());
-        SaveValue("SortIndex", sortIndex);
-        SaveValue("SortAscending", m_tlv->GetSortingColumn()->IsSortOrderAscending());
-    }
-
-    SaveValue("Col_Name_Width", m_tlv->GetColumn(TorrentListViewModel::Columns::Name)->GetWidth());
-    SaveValue("Col_QueuePosition_Width", m_tlv->GetColumn(TorrentListViewModel::Columns::QueuePosition)->GetWidth());
-    SaveValue("Col_Size_Width", m_tlv->GetColumn(TorrentListViewModel::Columns::Size)->GetWidth());
-    SaveValue("Col_Status_Width", m_tlv->GetColumn(TorrentListViewModel::Columns::Status)->GetWidth());
-    SaveValue("Col_Progress_Width", m_tlv->GetColumn(TorrentListViewModel::Columns::Progress)->GetWidth());
-    SaveValue("Col_ETA_Width", m_tlv->GetColumn(TorrentListViewModel::Columns::ETA)->GetWidth());
-    SaveValue("Col_DownloadSpeed_Width", m_tlv->GetColumn(TorrentListViewModel::Columns::DownloadSpeed)->GetWidth());
-    SaveValue("Col_UploadSpeed_Width", m_tlv->GetColumn(TorrentListViewModel::Columns::UploadSpeed)->GetWidth());
-    SaveValue("Col_Ratio_Width", m_tlv->GetColumn(TorrentListViewModel::Columns::Ratio)->GetWidth());
-    SaveValue("Col_Seeds_Width", m_tlv->GetColumn(TorrentListViewModel::Columns::Seeds)->GetWidth());
-    SaveValue("Col_Peers_Width", m_tlv->GetColumn(TorrentListViewModel::Columns::Peers)->GetWidth());
-    SaveValue("Col_AddedOn_Width", m_tlv->GetColumn(TorrentListViewModel::Columns::AddedOn)->GetWidth());
-    SaveValue("Col_CompletedOn_Width", m_tlv->GetColumn(TorrentListViewModel::Columns::CompletedOn)->GetWidth());
-}
-
-PersistentTorrentListView* pt::wxCreatePersistentObject(TorrentListView* lv)
-{
-    return new PersistentTorrentListView(lv);
-}
+#include "persistenttorrentlistview.hpp"
+
+#include "torrentlistview.hpp"
+#include "torrentlistviewmodel.hpp"
+
+#include "scaler.hpp"
+
+using pt::PersistentTorrentListView;
+using pt::TorrentListView;
+
+PersistentTorrentListView::PersistentTorrentListView(TorrentListView* tlv)
+    : wxPersistentWindow<TorrentListView>(tlv),
+    m_tlv(tlv)
+{
+}
+
+wxString PersistentTorrentListView::GetKind() const
+{
+    return "torrentViewCtrl";
+}
+
+bool PersistentTorrentListView::Restore()
+{
+    int sortIndex;
+    bool sortAscending;
+    int colWidth;
+
+    if (RestoreValue("Col_Name_Width", &colWidth))
+    {
+        m_tlv->GetColumn(TorrentListViewModel::Columns::Name)->SetWidth(SX(colWidth));
```

Comment:
We should not scale these values - they are scaled already.

Reply:
I store them removing the dpi scaling, better I think.

Reply:
I disagree - if we store it as-is we won't have to touch `PersistentTorrentListView` at all. This is preferable.

Reply:
I disagree as well. I'm changing my scaling all the time ( depending if i'm coding / watching stuff / browsing ). So I think storing the information in a non transform way is better. But you are the boss, so if you want it this is way, it will be :)

Reply:
Side note : would be a bit better to have a constant map with what we want to save instead of a list of it me think.

Reply:
That makes sense - I haven't thought about that. Your solution works for both cases so I think we should go with that then :)